File: backend/app/routes/notification_routes.py
# app/routes/notification_routes.py
from fastapi import APIRouter, Depends, HTTPException
from app.controllers.notification_controller import NotificationController
from app.middleware.auth import get_current_user

router = APIRouter(prefix="/notifications", tags=["Notifications"])


@router.get("/")
async def get_notifications(
    current_user=Depends(get_current_user), skip: int = 0, limit: int = 10
):
    """Retrieve user notifications"""
    try:
        return await NotificationController.get_notifications(
            current_user.id, skip, limit
        )
    except ValueError as e:
        raise HTTPException(status_code=400, detail=str(e))


@router.put("/mark-read/{notification_id}")
async def mark_notification_read(
    notification_id: int, current_user=Depends(get_current_user)
):
    """Mark a notification as read"""
    try:
        return await NotificationController.mark_notification_as_read(
            current_user.id, notification_id
        )
    except ValueError as e:
        raise HTTPException(status_code=400, detail=str(e))

File: backend/src/routes/notificationRoutes.js
// const express = require("express");
// const NotificationController = require("../controllers/notificationController");
// const { authenticate, authorize } = require("../middleware/authMiddleware");

// const router = express.Router();

// router.use(authenticate);

// router.get("/", NotificationController.getNotifications);
// router.patch("/mark-read", NotificationController.markAsRead);

// // Admin-only notification broadcast
// router.post(
//   "/broadcast",
//   authorize("ADMIN"),
//   NotificationController.sendNotification
// );

// module.exports = router;
// src/routes/notificationRoutes.js
const express = require("express");
const router = express.Router();
const NotificationController = require("../controllers/NotificationController");
const { authorize, authenticate } = require("../middleware/authMiddleware");

const controller = new NotificationController();
router.use(authenticate);

// Provider routes
router.post("/", authorize(["PROVIDER"]), controller.createNotification);

// User routes
router.get("/", authorize(["STUDENT"]), controller.getNotifications);
router.patch("/:id/read", authorize(["STUDENT"]), controller.markAsRead);
router.delete("/:id", authorize(["STUDENT"]), controller.deleteNotification);

module.exports = router;
